src/Polysemy/Internal/TH/Effect.hs

```diff
@@ -101,16 +99,19 @@ makeSem = genFreer True
 -- These restrictions may be removed in the future, depending on changes to
 -- the compiler.
 --
+-- Change in (TODO(Sandy): version): in case of GADTs, signatures now only use
```

Comment:
This isn't a breaking change, right? It should only affect haddock, so I don't think we need to do any serious sem versioning.

Reply:
It's a breaking change for users that quantified variables previously chosen by type constructor - example where explicit signature would be a problem is `get`, which changed from:
```hs
get :: forall a r. MemberWithError (State a) r => Sem r a 
```
to
```hs
get :: forall s r. MemberWithError (State s) r => Sem r s 
```
because data constructor uses `s` instead.